libraries/common/stream.effekt

```diff
@@ -238,6 +261,28 @@ def feed[T, R](list: List[T]) { reader: () => R / read[T] }: R = {
   }
 }
 
+def collectMap[K, V, R](compare: (K, K) => Ordering at {}) { stream: () => R / emit[(K, V)] }: (R, Map[K, V]) =
+  try {
+    (stream(), map::empty(compare))
+  } with emit[(K, V)] { case (k, v) =>
+    val (r, map) = resume(());
+    (r, map.put(k, v))
```

Comment:
For N inserts, this will allocate N frames. Alternatively we often use local mutable state to make this tail resumptive, which then runs in constant space.

Reply:
I wanted to keep the same style as the other `collectX` functions, but I agree.